sklearn/utils/_param_validation.py

```diff
@@ -392,14 +395,6 @@ class Interval(_Constraint):
     `[0, +∞) U {+∞}`.
     """
 
-    @validate_params(
-        {
-            "type": [type],
```

Comment:
Can we keep `validate_params` by using:

```python
"type": [Options(type, {Integral, Real}), StrOptions({"real_not_int"})],
```

Reply:
I removed this validation because it was a very bad idea from the start. You can't use it on all the constraint classes or you end up in infinite recursion. Also, it only works because it's using constraints that are written before in the code: put the definition of the StrOptions class below this one and it crashes.